### ISSUE TITLE
User Interface Issues

### ISSUE THREAD

**Issue by jerryfung110** [exanauts/ExaModels.jl#102]
1. Lack of Expression Support:

ExaModels does not support expressions like JuMP, requiring users to input long and complicated expressions simultaneously, making it unfriendly.

2. Iteration Limitation:

ExaModels cannot iterate a variable based on another variable within objective functions or constraints, expression cannot be represented concisely. 

3. Array Indexing Issue:

Arrays not generated by `ExaModels.variable()` cannot be indexed directly in objective functions and constraints. Attempting to do so results in errors like an invalid index: ExaModels.ParIndexed, because ExaModels creates an ExaModels.ParIndexed object instead of an Int, causing indexing issues.

4. Unclear Indexing in Constraints:

The indexing syntax in constraint! is unclear, which can lead to confusion and errors in defining constraints when users try to add expressions in `constraint!`

**Comment by jbcaillau** [comment]
@sshin23 AFAIR regarding arrays (declared as `variable`), it is not possible to use slices (only element wise operations combined with generators)

**Comment by jbcaillau** [comment]
@sshin23 yes. more generally being able to do some basic linear algebra (BLAS1-2) operations. eg, 

```julia
A = rand(10, 10)
x = variable(c, 1:10)
A[1, 1] * x # OK
```
